src/rtcp/packet.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * ```ts
+ * if (isRtcp(buffer)) {
+ *   console.log('it seems a valid RTCP packet');
+ * }
+ * ```
+ */
+export function isRtcp(buffer: Buffer): boolean
+{
+	const firstByte = buffer.readUInt8(0);
+
+	return (
+		Buffer.isBuffer(buffer) &&
+		buffer.length >= COMMON_HEADER_LENGTH &&
+		// DOC: https://tools.ietf.org/html/draft-ietf-avtcore-rfc5764-mux-fixes
+		(firstByte > 127 && firstByte < 192) &&
+		// RTCP Version must be 2.
+		(firstByte >> 6) === RTCP_VERSION &&
+		// RTCP packet types defined by IANA:
+		// http://www.iana.org/assignments/rtp-parameters/rtp-parameters.xhtml#rtp-parameters-4
+		// RFC 5761 (RTCP-mux) states this range for secure RTCP/RTP detection.
+		(buffer.readUInt8(1) >= 192 && buffer.readUInt8(1) <= 223)
+	);
+}
+
+export abstract class RtcpPacket
+{
+	// Buffer.
+	// @ts-ignore. 'buffer' has not initializer and is not assigned in constructor.
+	protected buffer: Buffer;
+	// RTCP Packet type.
+	private packetType: PacketType;
+
+	/**
+	 * @ignore
+	 *
+	 * @param PacketType.
+	 */
+	constructor(packetType: PacketType)
+	{
+		this.packetType = packetType;
+	}
+
+	/**
+	 * @ignore
+	 */
+	dump(): any
+	{
+		return {
+			version : this.getVersion(),
+			count   : this.getCount(),
+			length  : this.getLength(),
+			padding : this.getPadding()
+		};
+	}
+
+	/**
+	 * Get the internal buffer containing the serialized RTCP binary packet.
+	 */
+	abstract getBuffer(): Buffer;
+
+	/**
+	 * Get the RTCP version of the packet (always 2).
+	 */
+	getVersion(): number
+	{
+		return RTCP_VERSION;
+	}
+
+	/**
+	 * Set the RTCP version of the packet (always 2).
+	 */
+	private setVersion(): void
+	{
+		this.buffer.writeUInt8(this.buffer.readUInt8() | (RTCP_VERSION << 6), 0);
+	}
+
+	/**
+	 * Get the padding flag.
+	 */
+	getPadding(): boolean
+	{
+		return Boolean(this.buffer.readUInt8(0) & 0x20);
+	}
+
+	/**
+	 * Set the padding flag.
+	 */
+	setPadding(padding: boolean): void
+	{
+		// Update padding bit.
+		const bit = padding ? 1 : 0;
+
+		this.buffer.writeUInt8(this.buffer.readUInt8(0) | (bit << 5), 0);
+	}
+
+	/**
+	 * Get the RTCP header count value.
+	 */
+	getCount(): number
+	{
+		return this.buffer.readUInt8(0) & 0x1F;
+	}
+
+	/**
+	 * Set the RTCP header count value.
+	 */
+	protected setCount(count: number): void
+	{
+		this.buffer.writeUInt8(this.buffer.readUInt8() | (count & 0x1F), 0);
+	}
+
+	/**
+	 * Get the RTCP packet type.
+	 */
+	getPacketType(): PacketType
+	{
+		return this.buffer.readUInt8(1);
+	}
+
+	/**
+	 * Set the RTCP packet type.
+	 */
+	private setPacketType(count: PacketType): void
+	{
+		this.buffer.writeUInt8(count, 1);
+	}
+
+	/**
+	 * Get the RTCP packet length.
+	 */
+	getLength(): number
+	{
+		return this.buffer.readUInt16BE(2);
+	}
+
+	/**
+	 * Set the RTCP packet length.
+	 */
+	private setLength(length: number): void
+	{
+		this.buffer.writeUInt16BE(length, 2);
+	}
+
+	/**
+	 * Serialize RTCP packet into a new buffer.
+	 */
+	protected serialize(length: number): void
```

Comment:
The `serialize()` method is now public in `RtpPacket`. Can you check a look to its inline doc to see if the same may apply for RTCP?

Reply:
What do you mean? We can do `RTCP.serialize` public too.

Reply:
It's a protected method.

Reply:
This conversation was marked as resolved, but I don't agree. The `serialize()` method is public in `RtpPacket` for a reason:

```ts
/**
	 * Apply pending changes into the packet and serialize it into a new internal
	 * buffer (the one that [[getBuffer]] will later return).
	 *
	 * **NOTE:** In most cases there is no need to use this method. It must be
	 * called only if the application retrieves information from the packet (by
	 * calling [[getBuffer]], [[getPayload]], [[getExtension]], etc) and modifies
	 * the obtained buffers in place. However, it's recommended to use the
	 * existing setter methods instead ([[setPayload]], [[setExtension]], etc).
	 *
	 * @throws if invalid fields were previously added to the packet.
	 */
```

Reply:
Ok, I may be wrong since the user/app is not supposed to have direct access to the `RtcpPacket` class, right?